assets/course-theme/sidebar.js

```diff
@@ -0,0 +1,249 @@
+/**
+ * External dependencies
+ */
+import debounce from 'lodash/debounce';
+
+/**
+ * The last scroll top value.
+ *
+ * @member {number}
+ */
+let lastScrollTop = 0;
+
+/**
+ * Calculates the scroll delta.
+ */
+const getScrollDelta = () => {
+	const { scrollTop } = document.documentElement;
+	const delta = scrollTop - lastScrollTop;
+	lastScrollTop = Math.max( 0, scrollTop );
+	return delta;
+};
+
+/**
+ * Tells if the sidebar is supposed to be sticky.
+ *
+ * @return {boolean} True if it is sticky. False otherwise.
+ */
+const isStickySidebar = () =>
+	[ 'modern', 'video-full' ].some( ( templateName ) =>
+		document.body.classList.contains( `learning-mode--${ templateName }` )
+	);
+
+/**
+ * Sidebar margin top.
+ *
+ * @member {number}
+ */
+let sidebarMarginTop = 0;
+
+/**
+ * The sidebar DOM element.
+ *
+ * @member {HTMLElement}
+ */
+let sidebar = null;
+
+/**
+ * The header DOM element.
+ *
+ * @member {HTMLElement}
+ */
+let header = null;
+
+/**
+ * The clone of the sidebar DOM element. This is the sidebar element
+ * that user sees and interacts with.
+ *
+ * @member {HTMLElement}
+ */
+let stickySidebar = null;
+
+/**
+ * The featured video DOM element.
+ *
+ * @member {HTMLElement}
+ */
+let featuredVideo = null;
+
+/**
+ * Populates the DOM elements that we need.
+ */
+const queryDomElements = () => {
+	sidebar = document.querySelector( '.sensei-course-theme__sidebar' );
+	header = document.querySelector( '.sensei-course-theme__header' );
+	featuredVideo = document.querySelector(
+		'.sensei-course-theme-lesson-video'
+	);
+};
+
+/**
+ * Creates an exact copy of the sidebar DOM element
+ * and sets it's position to fixed. The original sidebar
+ * element is hidden by seting it's opacity to 0. The clone, "stickySidebar"
+ * is used to keep the sidebar sticky. We still need the original sidebar
+ * element because we use it's original position to calculate and decide
+ * where the stickySideber should be position at any given time.
+ *
+ * This can be called multiple times and if it detects an existing stickySidebar
+ * present in the DOM it will remove it and insert the new one.
+ */
+function preparestickySidebar() {
+	const sidebarRect = sidebar.getBoundingClientRect();
+	sidebarMarginTop = sidebar.style.marginTop
+		? parseInt( sidebar.style.marginTop, 10 )
+		: 0;
+	if ( stickySidebar?.remove ) {
+		stickySidebar.remove();
+	}
+	stickySidebar = sidebar.cloneNode( true );
+	stickySidebar.style.position = 'fixed';
```

Comment:
Would `position: sticky` be usable as a starting point? The browser would take away some of the positioning math and adjustments. Or is it harder to work with?

Reply:
Nah. I tried the `position: sticky` approach initially. I couldn't make it work properly.